README.md

```diff
@@ -3,6 +3,33 @@ Community driven Elm guide for JS people
 
 This guide will augment the official Elm docs as needed, for a smooth transitions for people coming from JavaScript.
 
+## Type Annotations
+
+Writing them is optional, but highly encouraged. Type annotations improves code by helping you think about what the function should be doing, and serve as compiler-verified documentation. In addition, if you ever want to publish a third-party library, you need type annotations.
+
+```elm
+-- A variable of type Int.
+answer : Int
+answer = 42
+
+-- An update function that takes 2 params (Action, Model) and returns a Model (last).
+update : Action -> Model -> Model
+-- could also be written as
+update : (Action, Model) -> Model
+```
+
+In JavaScript, params are handled at the same time. In Elm, they are curried. For example:
```

Comment:
IMHO, if you want to talk about currying you should really introduce it as a concept. It's super simple once you get it, but it looks strange when you first see and it sounds like it might be a hard thing because it has a name.